daemon/graphdriver/copy/copy.go

```diff
@@ -12,13 +12,19 @@ package copy
 import "C"
 import (
 	"container/list"
+	"context"
 	"fmt"
+	"hash/fnv"
 	"io"
 	"os"
 	"path/filepath"
+	"sync"
 	"syscall"
 	"time"
 
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/syncmap"
```

Comment:
a nitpick: AFAIK in go 1.9 you can use `sync.Map` from the standard `sync` package (and currently Moby uses and depends on Go 1.9). https://golang.org/pkg/sync/#Map